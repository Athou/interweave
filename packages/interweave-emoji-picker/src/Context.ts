/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import React from 'react';

export default React.createContext({
  classNames: {},
  messages: {},
});
