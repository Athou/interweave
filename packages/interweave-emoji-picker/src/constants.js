/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 * @flow
 */

// Group order must match emojibase
export const GROUPS: string[] = [
  'smileysPeople',
  'animalsNature',
  'foodDrink',
  'travelPlaces',
  'activities',
  'objects',
  'symbols',
  'flags',
];

export const GROUP_ICONS: { [key: string]: string } = {
  smileysPeople: '😃',
  animalsNature: '🌿',
  foodDrink: '🍎',
  travelPlaces: '🗺️',
  activities: '⚽️',
  objects: '📘',
  symbols: '⛔️',
  flags: '🏴',
};

export const DEFAULT_GROUP: string = 'smileysPeople';

export const SCROLL_DEBOUNCE: number = 50;

export const SEARCH_THROTTLE: number = 150;
