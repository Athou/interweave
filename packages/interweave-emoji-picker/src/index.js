/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 * @flow
 */

import Picker from './Picker';

export default Picker;
