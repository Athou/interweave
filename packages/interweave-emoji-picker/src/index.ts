/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import withEmojiPickerContext, { Context, ContextProps } from './Context';
import Picker, { PickerProps } from './Picker';

export { withEmojiPickerContext, ContextProps, PickerProps };

export * from './shapes';

export * from './types';

export default Picker;
