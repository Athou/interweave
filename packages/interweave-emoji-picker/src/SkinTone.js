/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 * @flow
 */

import React from 'react';
import PropTypes from 'prop-types';
import { SKIN_COLORS } from './constants';

type SkinToneProps = {
  activeSkinTone: string,
  onSelect: (skinTone: string, e: *) => void,
  skinTone: string,
};

export default class SkinTone extends React.PureComponent<SkinToneProps> {
  static contextTypes = {
    classNames: PropTypes.objectOf(PropTypes.string),
  };

  static propTypes = {
    activeSkinTone: PropTypes.string.isRequired,
    skinTone: PropTypes.string.isRequired,
    onSelect: PropTypes.func.isRequired,
  };

  /**
   * Triggered when the button is clicked.
   */
  handleClick = (e: SyntheticMouseEvent<HTMLButtonElement>) => {
    e.stopPropagation();

    this.props.onSelect(this.props.skinTone, e);
  };

  render() {
    const { activeSkinTone, skinTone } = this.props;
    const { classNames } = this.context;
    const className = [classNames.skinTone];
    const color = SKIN_COLORS[skinTone];

    if (skinTone === activeSkinTone) {
      className.push(classNames.skinToneActive);
    }

    return (
      <button
        className={className.join(' ')}
        style={{ backgroundColor: color, borderColor: color }}
        type="button"
        onClick={this.handleClick}
      >
        &nbsp;
      </button>
    );
  }
}
