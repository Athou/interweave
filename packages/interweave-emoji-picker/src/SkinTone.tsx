/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import React from 'react';
import PropTypes from 'prop-types';
import { SKIN_COLORS } from './constants';

export interface SkinToneProps {
  active: boolean;
  children?: React.ReactNode;
  onSelect: (skinTone: string, event: React.MouseEvent<HTMLButtonElement>) => void;
  skinTone: string;
}

export default class SkinTone extends React.PureComponent<SkinToneProps> {
  static contextTypes = {
    classNames: PropTypes.objectOf(PropTypes.string),
  };

  static propTypes = {
    active: PropTypes.bool.isRequired,
    children: PropTypes.node,
    onSelect: PropTypes.func.isRequired,
    skinTone: PropTypes.string.isRequired,
  };

  static defaultProps = {
    children: null,
  };

  /**
   * Triggered when the button is clicked.
   */
  private handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.stopPropagation();

    this.props.onSelect(this.props.skinTone, e);
  };

  render() {
    const { active, children, skinTone } = this.props;
    const { classNames } = this.context;
    const className = [classNames.skinTone];
    const color = SKIN_COLORS[skinTone];

    if (active) {
      className.push(classNames.skinToneActive);
    }

    return (
      <button
        className={className.join(' ')}
        style={{ backgroundColor: color, borderColor: color, color }}
        type="button"
        onClick={this.handleClick}
      >
        {children || ' '}
      </button>
    );
  }
}
