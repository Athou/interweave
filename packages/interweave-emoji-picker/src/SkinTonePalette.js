/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 * @flow
 */

import React from 'react';
import PropTypes from 'prop-types';
import SkinTone from './SkinTone';
import { SKIN_TONES } from './constants';

type SkinTonePaletteProps = {
  activeSkinTone: string,
  onSelect: (skinTone: string) => void,
};

export default class SkinTonePalette extends React.PureComponent<SkinTonePaletteProps> {
  static contextTypes = {
    classNames: PropTypes.objectOf(PropTypes.string),
  };

  static propTypes = {
    activeSkinTone: PropTypes.string.isRequired,
    onSelect: PropTypes.func.isRequired,
  };

  render() {
    const { activeSkinTone, onSelect } = this.props;
    const { classNames } = this.context;

    return (
      <div className={classNames.skinTones}>
        {SKIN_TONES.map(skinTone => (
          <SkinTone
            key={skinTone}
            activeSkinTone={activeSkinTone}
            skinTone={skinTone}
            onSelect={onSelect}
          />
        ))}
      </div>
    );
  }
}
