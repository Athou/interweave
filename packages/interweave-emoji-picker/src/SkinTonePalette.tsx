/**
 * @copyright   2016, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import React from 'react';
import PropTypes from 'prop-types';
import SkinTone from './SkinTone';
import { SKIN_TONES } from './constants';
import { SkinTone as SkinToneType } from './types';

export interface SkinTonePaletteProps {
  activeSkinTone: SkinToneType;
  icons: { [key: string]: React.ReactNode };
  onSelect: (skinTone: SkinToneType, event: React.MouseEvent<HTMLButtonElement>) => void;
}

export default class SkinTonePalette extends React.PureComponent<SkinTonePaletteProps> {
  static contextTypes = {
    classNames: PropTypes.objectOf(PropTypes.string),
  };

  static propTypes = {
    activeSkinTone: PropTypes.string.isRequired,
    icons: PropTypes.objectOf(PropTypes.node).isRequired,
    onSelect: PropTypes.func.isRequired,
  };

  render() {
    const { activeSkinTone, icons, onSelect } = this.props;
    const { classNames } = this.context;

    return (
      <div className={classNames.skinTones}>
        {SKIN_TONES.map(skinTone => (
          <SkinTone
            key={skinTone}
            active={activeSkinTone === skinTone}
            skinTone={skinTone}
            onSelect={onSelect}
          >
            {icons[skinTone] || null}
          </SkinTone>
        ))}
      </div>
    );
  }
}
