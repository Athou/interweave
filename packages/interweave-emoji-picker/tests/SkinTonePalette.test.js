import React from 'react';
import { shallow } from 'enzyme';
import { SKIN_TONES, SKIN_NONE, SKIN_DARK } from '../src/constants';
import SkinTone from '../src/SkinTone';
import SkinTonePalette from '../src/SkinTonePalette';
import { PICKER_CONTEXT } from './mocks';

describe('<SkinTonePalette />', () => {
  const context = PICKER_CONTEXT;

  const props = {
    activeSkinTone: SKIN_NONE,
    icons: {},
    onSelect() {},
  };

  it('renders a palette', () => {
    const wrapper = shallow(<SkinTonePalette {...props} />, { context });

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a node for each skin tone', () => {
    const wrapper = shallow(<SkinTonePalette {...props} />, { context });

    expect(wrapper.find(SkinTone)).toHaveLength(SKIN_TONES.length);
  });

  it('can customize class name', () => {
    const wrapper = shallow(<SkinTonePalette {...props} />, {
      context: {
        ...context,
        classNames: {
          ...context.classNames,
          skinTones: 'test-skin-tones',
        },
      },
    });

    expect(wrapper.prop('className')).toBe('test-skin-tones');
  });

  it('passes `onSelect` to skin tones', () => {
    const wrapper = shallow(<SkinTonePalette {...props} />, { context });

    expect(wrapper.find(SkinTone).at(0).prop('onSelect')).toBe(props.onSelect);
  });

  it('sets active skin tone', () => {
    const wrapper = shallow(<SkinTonePalette {...props} activeSkinTone={SKIN_DARK} />, { context });
    const tones = wrapper.find(SkinTone);

    expect(tones.filterWhere(node => node.key() === SKIN_NONE).prop('active')).toBe(false);
    expect(tones.filterWhere(node => node.key() === SKIN_DARK).prop('active')).toBe(true);
  });
});
