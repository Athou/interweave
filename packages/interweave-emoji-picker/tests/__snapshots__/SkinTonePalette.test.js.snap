// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkinTonePalette /> renders a palette 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkinTonePalette
    activeSkinTone="none"
    icons={Object {}}
    onSelect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SkinTone
          active={true}
          onSelect={[Function]}
          skinTone="none"
        />,
        <SkinTone
          active={false}
          onSelect={[Function]}
          skinTone="light"
        />,
        <SkinTone
          active={false}
          onSelect={[Function]}
          skinTone="mediumLight"
        />,
        <SkinTone
          active={false}
          onSelect={[Function]}
          skinTone="medium"
        />,
        <SkinTone
          active={false}
          onSelect={[Function]}
          skinTone="mediumDark"
        />,
        <SkinTone
          active={false}
          onSelect={[Function]}
          skinTone="dark"
        />,
      ],
      "className": "interweave-picker__skin-tones",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "none",
        "nodeType": "class",
        "props": Object {
          "active": true,
          "children": null,
          "onSelect": [Function],
          "skinTone": "none",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "light",
        "nodeType": "class",
        "props": Object {
          "active": false,
          "children": null,
          "onSelect": [Function],
          "skinTone": "light",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "mediumLight",
        "nodeType": "class",
        "props": Object {
          "active": false,
          "children": null,
          "onSelect": [Function],
          "skinTone": "mediumLight",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "medium",
        "nodeType": "class",
        "props": Object {
          "active": false,
          "children": null,
          "onSelect": [Function],
          "skinTone": "medium",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "mediumDark",
        "nodeType": "class",
        "props": Object {
          "active": false,
          "children": null,
          "onSelect": [Function],
          "skinTone": "mediumDark",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "dark",
        "nodeType": "class",
        "props": Object {
          "active": false,
          "children": null,
          "onSelect": [Function],
          "skinTone": "dark",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SkinTone
            active={true}
            onSelect={[Function]}
            skinTone="none"
          />,
          <SkinTone
            active={false}
            onSelect={[Function]}
            skinTone="light"
          />,
          <SkinTone
            active={false}
            onSelect={[Function]}
            skinTone="mediumLight"
          />,
          <SkinTone
            active={false}
            onSelect={[Function]}
            skinTone="medium"
          />,
          <SkinTone
            active={false}
            onSelect={[Function]}
            skinTone="mediumDark"
          />,
          <SkinTone
            active={false}
            onSelect={[Function]}
            skinTone="dark"
          />,
        ],
        "className": "interweave-picker__skin-tones",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "none",
          "nodeType": "class",
          "props": Object {
            "active": true,
            "children": null,
            "onSelect": [Function],
            "skinTone": "none",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "light",
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": null,
            "onSelect": [Function],
            "skinTone": "light",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "mediumLight",
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": null,
            "onSelect": [Function],
            "skinTone": "mediumLight",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "medium",
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": null,
            "onSelect": [Function],
            "skinTone": "medium",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "mediumDark",
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": null,
            "onSelect": [Function],
            "skinTone": "mediumDark",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "dark",
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": null,
            "onSelect": [Function],
            "skinTone": "dark",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "classNames": Object {
        "emoji": "interweave-picker__emoji",
        "emojiActive": "interweave-picker__emoji--active",
        "emojis": "interweave-picker__emojis",
        "emojisBody": "interweave-picker__emojis-body",
        "emojisContainer": "interweave-picker__emojis-container",
        "emojisHeader": "interweave-picker__emojis-header",
        "emojisRow": "interweave-picker__emojis-row",
        "emojisSection": "interweave-picker__emojis-section",
        "group": "interweave-picker__group",
        "groupActive": "interweave-picker__group--active",
        "groups": "interweave-picker__groups",
        "groupsList": "interweave-picker__groups-list",
        "noPreview": "interweave-picker__no-preview",
        "noResults": "interweave-picker__no-results",
        "picker": "interweave-picker__picker",
        "pickerVirtual": "interweave-picker__picker--virtual",
        "preview": "interweave-picker__preview",
        "previewContent": "interweave-picker__preview-content",
        "previewEmoji": "interweave-picker__preview-emoji",
        "previewSubtitle": "interweave-picker__preview-subtitle",
        "previewTitle": "interweave-picker__preview-title",
        "search": "interweave-picker__search",
        "searchInput": "interweave-picker__search-input",
        "skinTone": "interweave-picker__skin-tone",
        "skinToneActive": "interweave-picker__skin-tone--active",
        "skinTones": "interweave-picker__skin-tones",
      },
      "messages": Object {
        "activities": "Activities",
        "animalsNature": "Animals & Nature",
        "flags": "Flags",
        "foodDrink": "Food & Drink",
        "frequentlyUsed": "Frequently Used",
        "noGroups": "All Emojis",
        "noPreview": "No emoji to display",
        "noResults": "No results",
        "objects": "Objects",
        "recentlyUsed": "Recently Used",
        "search": "Search",
        "searchA11y": "Search for emojis by keyword",
        "searchResults": "Search Results",
        "smileysPeople": "Smileys & People",
        "symbols": "Symbols",
        "travelPlaces": "Travel & Places",
      },
    },
  },
}
`;
